platform/ios/MGLAccountManager.m

```diff
@@ -0,0 +1,50 @@
+#import <Foundation/Foundation.h>
+
+#import "MGLAccountManager.h"
+#import "NSProcessInfo+MGLAdditions.h"
+#import "MGLMapboxEvents.h"
+
+@interface MGLAccountManager()
+
+@property (atomic) NSString *accessToken;
+
+@end
+
+
+@implementation MGLAccountManager
+
+static MGLAccountManager *_sharedManager;
+
+// Can be called from any thread. Called implicitly from any
+// public class convenience methods.
+//
++ (instancetype) sharedInstanceWithAccessToken:(NSString *)token {
```

Comment:
This method’s name is misleading. It implies that there can be more than one shared instance, that we’re mapping access tokens to shared instances somehow. What happens when this class begins to manage more than just an access token? `+sharedInstanceWithUserName:accessToken:expiryDate:cachingPragma:`?

I still think the two options in https://github.com/mapbox/mapbox-gl-native/pull/1329#discussion_r29024497 are the best ways forward.